lib/https.js

```diff
@@ -58,7 +58,22 @@ function createConnection(port, host, options) {
   }
 
   debug('createConnection', options);
-  return tls.connect(options);
+
+  var self = this;
+  if (options._agentKey && this._tlsSessions[options._agentKey]) {
+    debug('reuse session for %j', options._agentKey);
+    options = util._extend({
+      session: this._tlsSessions[options._agentKey]
```

Comment:
Yeah, but we usually try to avoid changing the objects that are passed to the function.